helios-client/src/main/java/com/spotify/helios/common/JobValidator.java

```diff
@@ -208,9 +209,20 @@ public JobValidator(final boolean shouldValidateJobHash,
       }
     }
 
+    errors.addAll(validateGracePeriodAndExternalPorts(job));
+
     return errors;
   }
 
+  private Set<String> validateGracePeriodAndExternalPorts(final Job job) {
+    final Integer gracePeriod = job.getGracePeriod();
+    if (gracePeriod != null && gracePeriod > 0 && job.hasExternalPorts()) {
+      return singleton(
+          "Job has a grace period and static/external ports. Please choose one or the other.");
```

Comment:
How about adding a brief description of the reason only one of the two can be chosen? Maybe just as simple as "Please choose one or the other, as the two conflict with each other".

Reply:
Thanks. Updated with "This configuration will prevent new containers from deploying during the gracePeriod because of port conflicts"